pkg/daemon/update.go

```diff
@@ -1502,12 +1502,22 @@ func (dn *Daemon) writeUnits(units []ign3types.Unit) error {
 	for _, u := range units {
 		// write the dropin to disk
 		for i := range u.Dropins {
+			dpath := filepath.Join(pathSystemd, u.Name+".d", u.Dropins[i].Name)
 			if u.Dropins[i].Contents == nil || *u.Dropins[i].Contents == "" {
 				glog.Infof("Dropin for %s has no content, skipping write", u.Dropins[i].Name)
+				if _, err := os.Stat(dpath); err != nil {
+					if _, ok := err.(*os.PathError); ok {
```

Comment:
Minor point: correct me if I'm wrong but I think you are checking to see whether the file exists, and if it doesn't, just continue right?

For situations that err is not PathError (or I guess if we want to be specific, IsNotExist) should we just return the error here? Otherwise it will go to the remove below and may or may not cause other issues.

Reply:
Updated to `os.IsNotExist`.
 
I agree the logic is a little wonky...so changed. 

The check I wanted was:
- look to see if the file is missing, if so continue
- log a message about removing (@crawford asked for the message)
- then remove